src/Multipart/AbstractUploader.php

```diff
@@ -0,0 +1,455 @@
+<?php
+namespace Aws\Multipart;
+
+use Aws\AwsClientInterface as Client;
+use Aws\CommandInterface;
+use Aws\CommandPool;
+use Aws\Exception\AwsException;
+use Aws\Exception\MultipartUploadException;
+use Aws\Result;
+use Aws\ResultInterface;
+use GuzzleHttp\Promise;
+use GuzzleHttp\Promise\PromiseInterface;
+use GuzzleHttp\Psr7;
+use InvalidArgumentException as IAE;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\StreamableInterface as Stream;
+
+/**
+ * Encapsulates the execution of a multipart upload to S3 or Glacier.
+ */
+abstract class AbstractUploader
```

Comment:
Aborting as a cancel action doesn't make sense. It's a decision you make. For example, you might create a Uploader object for the sole purpose of aborting the upload. That is why I made upload/uploadAsync and abort/abortAsync. Trying to combine them is very weird. I at least chose the same convention as our client objects though, with the async suffix.